notify/telegram.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env sh
+
+#Support Telegram Bots
+
+#TELEGRAM_BOT_APITOKEN=""
+#TELEGRAM_BOT_CHATID=""
+
+telegram_send() {
+  _subject="$1"
+  _content="$2"
+  _statusCode="$3" #0: success, 1: error 2($RENEW_SKIP): skipped
+  _debug "_statusCode" "$_statusCode"
+
+  TELEGRAM_BOT_APITOKEN="${TELEGRAM_BOT_APITOKEN:-$(_readaccountconf_mutable TELEGRAM_BOT_APITOKEN)}"
```

Comment:
see : https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh#L22

please use "_getdeployconf" and "_savedeployconf "

Reply:
No problem, I can do that - but, to be clear, that's different to all of the other [existing notify hooks](https://github.com/acmesh-official/acme.sh/tree/dev/notify), is that intentional? (should they be converted to use `_{save,get}deployconf` too? so this would just be the first notify hook to use them?)

Reply:
Yes,  the `_{save,get}deployconf` is the newer way to get and set deploy options.

Most of the existing hooks are not updated yet.

Reply:
After reading the main shell script yesterday, I now see the intention of the various `_{save,get}*conf` functions... however, is a new set of functions required for notify hooks, since they're more of a global setting, not a per-domain setting? As it stands, the `deployconf` variants save to the `$domain/$domain.conf` file. Unless I'm mistaken, notification hooks are called for all domains after `--cron`, no?

Reply:
(I can also see a world where each domain might have different notification settings, which are processed after a renewal attempt, instead of one global notification.. while more complicated, that might have merit?)

Reply:
> Unless I'm mistaken, notification hooks are called for all domains after `--cron`, no?

[sorry for the multiple messages, I should stop and finish reading before posting!]

Turns out I am mistaken, I'd completely missed the `$NOTIFY_MODE` setting... however, that does indicate that perhaps there would be some unintended behaviour if notify hook specific settings are saved in a domain file. When `$NOTIFY_MODE` is set to `0` for bulk notifications (which is the default), then I think the settings of the _last_ domain renewed would be used for the bulk notification? The notification hooks would need to check `$NOTIFY_MODE` as well, and choose to load from the domain config, or the account config.

hopefully this discussion is helpful, and not unwelcome. I would be keen to put together a PR that addresses these issues once this one is merged if you like.

Reply:
I'm so sorry. I missed up when I was reviewing multiple PRs.  please forget the `_{save,get}deployconf`, it's for deploy hooks only.

It's ok for you notification hook to use account options.

Reply:
No problem - however, I do think that there might actually be merit in having per domain notifications, and I can see it being relatively easy to implement.

Thanks for the merge!

Reply:
>>  No problem - however, I do think that there might actually be merit in having per domain notifications, and I can see it being relatively easy to implement.

I don't think so.   it will make the notifications more complicated for non-expert users.
We can think about it if there are too many users require. but for now, I don't hear anyone need that.